src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -140,6 +141,62 @@ Convolution<
   padding = ann::Padding<>(padWLeft, padWRight, padHTop, padHBottom);
 }
 
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::Convolution(
+    const Convolution& network) :
+    inSize(network.inSize),
+    outSize(network.outSize),
+    kernelWidth(network.kernelWidth),
+    kernelHeight(network.kernelHeight),
+    strideWidth(network.strideWidth),
+    strideHeight(network.strideHeight),
+    padWLeft(network.padWLeft),
+    padWRight(network.padWRight),
+    padHBottom(network.padHBottom),
+    padHTop(network.padHTop),
+    inputWidth(network.inputWidth),
+    inputHeight(network.inputHeight),
+    outputWidth(network.outputWidth),
+    outputHeight(network.outputHeight),
+    paddingType(network.paddingType),
+    weight(network.weight),
+    bias(network.bias)
+{
+  weights.set_size((outSize * inSize * kernelWidth * kernelHeight) + outSize,
+      1);
+
+  // Transform paddingType to lowercase.
+  std::string paddingTypeLow = paddingType;
```

Comment:
Since we use this at various places now it would make sense to put it into: https://github.com/mlpack/mlpack/tree/master/src/mlpack/core/util. Let me know what you think.

Reply:
Sure nice idea. I will add it thanks for suggestion @zoq

Reply:
Hey, @zoq I did this in new PR #2377 as this one already containing a lot of code to be reviewed.